### PR TITLE
Sn 2245 - google backup integration

### PR DESCRIPTION
**Pull request by mrZizik**

### CODE REVIEW

File: core_db/schemas/jp.co.soramitsu.core_db.AppDatabase/70.json
Comment (AmadeyKuspakov) [inline core_db/schemas/jp.co.soramitsu.core_db.AppDatabase/70.json:1]: Left after DB migration

File: common/src/main/java/jp/co/soramitsu/common/base/SoraBaseFragment.kt [hunk 1]
```diff
@@ -85,6 +85,8 @@ abstract class SoraBaseFragment<T : BaseViewModel> : Fragment() {
     @Inject
     lateinit var debounceClickHandler: DebounceClickHandler
 
+    var navController: NavHostController? = null
```
Comment (AmadeyKuspakov) [inline common/src/main/java/jp/co/soramitsu/common/base/SoraBaseFragment.kt:88]: Why do we want to store this? Seems like a possible memory leak

File: feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsFragment.kt [hunk 2]
```diff
@@ -61,6 +64,18 @@ class AccountDetailsFragment : SoraBaseFragment<AccountDetailsViewModel>() {
     @Inject
     lateinit var vmf: AccountDetailsViewModel.AccountDetailsViewModelFactory
 
+    private val launcher =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            if (result.resultCode != Activity.RESULT_OK) {
+                Toast.makeText(requireActivity(), "Google signin failed", Toast.LENGTH_SHORT)
```
Comment (AmadeyKuspakov) [inline feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsFragment.kt:70]: String res

File: feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/backup/BackupScreen.kt [hunk 3]
```diff
@@ -190,7 +241,7 @@ private fun PreviewBackup() {
                 mnemonicWords = List(15) { p -> "zxcvb n ${p + 1}" },
             ),
             onButtonPressed = {},
-            onSkipButtonPressed = {}
+//            onSkipButtonPressed = {}
```
Comment (AmadeyKuspakov) [inline feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/backup/BackupScreen.kt:244]: Remove comments

File: feature_wallet_impl/build.gradle [hunk 4]
```diff
@@ -142,6 +142,9 @@ dependencies {
 
     implementation xNetworkingDep
 
+    implementation guavaDep
```
Comment (AmadeyKuspakov) [inline feature_wallet_impl/build.gradle:145]: Guava

File: build.gradle [hunk 5]
```diff
@@ -51,7 +51,8 @@ buildscript {
             accompanist            : '0.30.1',
             xserializationjson     : '1.5.0',
             composeMaterialDialog  : '0.9.0',
-            ktor                   : '2.2.3'
+            ktor                   : '2.2.3',
+            guava                  : '31.0.1-android'
```
Comment (AmadeyKuspakov) [inline build.gradle:55]: Guava

File: common/src/main/java/jp/co/soramitsu/common/presentation/compose/components/CheckboxButton.kt [hunk 6]
```diff
@@ -0,0 +1,62 @@
+package jp.co.soramitsu.common.presentation.compose.components
```
Comment (AmadeyKuspakov) [inline common/src/main/java/jp/co/soramitsu/common/presentation/compose/components/CheckboxButton.kt:1]: Licence template

File: common/src/main/res/values-in/values-id/strings.xml [hunk 7]
```diff
@@ -0,0 +1,507 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (AmadeyKuspakov) [inline common/src/main/res/values-in/values-id/strings.xml:1]: Didn't we already have value-in?

File: common/src/main/res/values-de/values-de-rDE/strings.xml
Comment (AmadeyKuspakov) [inline common/src/main/res/values-de/values-de-rDE/strings.xml:1]: Didn't we already have value-de-rDE?

File: feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/OnboardingActivity.kt
Comment (AmadeyKuspakov) [inline feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/OnboardingActivity.kt:1]: Licence template

File: feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/OnboardingFlowState.kt
Comment (AmadeyKuspakov) [inline feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/OnboardingFlowState.kt:1]: Licence template

File: feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/backup_password/BackupPasswordScreen.kt [hunk 8]
```diff
@@ -0,0 +1,136 @@
+/**
```
Comment (AmadeyKuspakov) [inline feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/backup_password/BackupPasswordScreen.kt:1]: Licence template has changed

File: feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsFragment.kt
Comment (AmadeyKuspakov) [inline feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsFragment.kt:1]: Licence template

File: feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsScreen.kt
Comment (AmadeyKuspakov) [inline feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsScreen.kt:1]: Licence template

File: feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsViewModel.kt
Comment (AmadeyKuspakov) [inline feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsViewModel.kt:1]: Licence template

File: feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/backup/BackupScreen.kt
Comment (AmadeyKuspakov) [inline feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/backup/BackupScreen.kt:1]: Licence template

File: feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/protection/ExportProtectionScreen.kt
Comment (AmadeyKuspakov) [inline feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/protection/ExportProtectionScreen.kt:1]: Licence template

File: feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsViewModel.kt [hunk 9]
```diff
@@ -149,4 +174,125 @@ class AccountDetailsViewModel @AssistedInject constructor(
...
+        _createBackupPasswordState.value?.let {
+            val newWarningState = !it.warningIsSelected
+            _createBackupPasswordState.value = it.copy(
+                warningIsSelected = newWarningState,
+                setPasswordButtonIsEnabled = newWarningState && it.password.value.text == it.passwordConfirmation.value.text && it.password.value.text.isPasswordSecure()
+            )
+        }
+    }
+
+    fun onBackupPasswordClicked(
```
Comment (AmadeyKuspakov) [inline feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsViewModel.kt:217]: Seems like a hard-to-unit-test method

File: feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsViewModel.kt [hunk 10]
```diff
@@ -149,4 +174,125 @@ class AccountDetailsViewModel @AssistedInject constructor(
...
+                        _accountDetailsScreenState.value =
+                            accountDetailsScreenState.copy(fileId = fileID)
+                        navController.popBackStack()
+                    }
+                }
+            }
+        }
+    }
+
+    fun onBackupClicked(
```
Comment (AmadeyKuspakov) [inline feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsViewModel.kt:251]: Seems like a hard-to-unit-test method

File: feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsViewModel.kt [hunk 11]
```diff
@@ -149,4 +174,125 @@ class AccountDetailsViewModel @AssistedInject constructor(
...
+
+                        navController.navigate(AccountDetailsRoutes.BACKUP_ACCOUNT)
+                        _accountDetailsScreenState.value = it.copy(isBackupLoading = false)
+                    }
+                }
+            }
+        }
+    }
+
+    fun onSuccessfulGoogleSignin(activity: Activity, navController: NavController) {
```
Comment (AmadeyKuspakov) [inline feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsViewModel.kt:283]: Seems like a hard-to-unit-test method

File: feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsViewModel.kt [hunk 12]
```diff
@@ -149,4 +174,125 @@ class AccountDetailsViewModel @AssistedInject constructor(
...
+        }
+    }
+
+    fun onBackupPasswordClicked(
+        activity: Activity,
+        navController: NavController
+    ) {
+        _createBackupPasswordState.value?.let { createBackupPasswordState ->
+            _createBackupPasswordState.value = createBackupPasswordState.copy(isLoading = true)
+            viewModelScope.launch(Dispatchers.IO) {
```
Comment (AmadeyKuspakov) [inline feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsViewModel.kt:223]: Will not possible to write unit tests, it's better to lean towards CoroutineManager usage

File: feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsViewModel.kt [hunk 13]
```diff
@@ -149,4 +174,125 @@ class AccountDetailsViewModel @AssistedInject constructor(
...
+                            mnemonic,
+                            "",
+                            CryptoType.SR25519
+                        ),
+                        createBackupPasswordState.password.value.text
+                    )
+
+                    interactor.updateBackupFileId(accountDetailsScreenState.address, fileID)
+
+                    withContext(Dispatchers.Main) {
```
Comment (AmadeyKuspakov) [inline feature_multiaccount_impl/src/main/java/jp/co/soramitsu/feature_multiaccount_impl/presentation/export_account/account_details/AccountDetailsViewModel.kt:241]: Will not possible to write unit tests, it's better to lean towards CoroutineManager usage

**Review by VitalyPeryatin — APPROVED** [review]

**Review by arvifox — APPROVED** [review]

**Pull request merged**